Gems/Atom/RPI/Code/Tests/Model/ModelTests.cpp

```diff
@@ -1031,42 +1035,76 @@ namespace UnitTest
     static constexpr inline auto minmaxElement = AZStd::minmax_element(begin(TwoSeparatedPlanesIndices), end(TwoSeparatedPlanesIndices));
     static_assert(*minmaxElement.second == (TwoSeparatedPlanesPositions.size() / 3) - 1);
 
-    template<class x> class TD;
     class TestMesh
     {
     public:
+        TestMesh() = default;
+
         TestMesh(const float* positions, size_t positionCount, const uint32_t* indices, size_t indicesCount)
         {
             AZ::RPI::ModelLodAssetCreator lodCreator;
+            Begin(lodCreator);
+            Add(lodCreator, positions, positionCount, /*positionOffset=*/0, indices, indicesCount, /*indexOffset=*/0);
+            End(lodCreator);
+        }
+
+        // initiate the asset lod creation process (note: End must be called after meshes have been added).
+        void Begin(AZ::RPI::ModelLodAssetCreator& lodCreator)
+        {
             lodCreator.Begin(AZ::Data::AssetId(AZ::Uuid::CreateRandom()));
+        }
 
+        // add a sub mesh and reuse existing position/index buffer (be very careful with the offsets used)
+        void Add(
+            AZ::RPI::ModelLodAssetCreator& lodCreator,
+            const float* positions,
+            size_t positionCount,
+            size_t positionOffset,
+            AZ::Data::Asset<AZ::RPI::BufferAsset> positionBuffer,
+            const uint32_t* indices,
+            size_t indexCount,
+            size_t indexOffset,
+            AZ::Data::Asset<AZ::RPI::BufferAsset> indexBuffer)
+        {
             lodCreator.BeginMesh();
-            lodCreator.SetMeshAabb(AZ::Aabb::CreateFromMinMax({-1.0f, -1.0f, -0.5f}, {1.0f, 1.0f, 0.5f}));
+            lodCreator.SetMeshAabb(AZ::Aabb::CreateFromMinMax({ -1.0f, -1.0f, -0.5f }, { 1.0f, 1.0f, 0.5f }));
             lodCreator.SetMeshMaterialSlot(AZ::Sfmt::GetInstance().Rand32());
 
-            {
-                AZ::Data::Asset<AZ::RPI::BufferAsset> indexBuffer = BuildTestBuffer(static_cast<uint32_t>(indicesCount), sizeof(uint32_t));
-                AZStd::copy(indices, indices + indicesCount, reinterpret_cast<uint32_t*>(const_cast<uint8_t*>(indexBuffer->GetBuffer().data())));
-                lodCreator.SetMeshIndexBuffer({
-                    indexBuffer,
-                    AZ::RHI::BufferViewDescriptor::CreateStructured(0, static_cast<uint32_t>(indicesCount), sizeof(uint32_t))
-                });
-            }
+            AZStd::copy(indices, indices + indexCount, reinterpret_cast<uint32_t*>(const_cast<uint8_t*>(indexBuffer->GetBuffer().data())));
```

Comment:
Hmm you could be right... but I'll have to check why the test passes and gets the right data... I'll double check tomorrow

Reply:
You're overwriting the first few vertices of the cube with the vertices from the quad. So the resulting model still has your offset quad, plus some vertices from the cube. So the intersection test will pass, since the quad is there.

Reply:
You're quite right, good spot! This is now fixed and improved by this commit - https://github.com/o3de/o3de/pull/5447/commits/b6dfd9e0f68494dee0c2a9103906baea168a0666